odoo/orm/models.py

```diff
@@ -247,23 +247,21 @@ def add_default(name, field):
                     setattr(self, name, field)
                     field.__set_name__(self, name)
 
-            add('id', Id(automatic=True))
-            add_default('display_name', Char(
-                string='Display Name', automatic=True,
-                compute='_compute_display_name',
-                search='_search_display_name',
-            ))
+            # make sure `id` is not overwritten
+            id_field = getattr(self, 'id', None)
+            if not (id_field is not None and isinstance(id_field, Id)):
+                raise TypeError(f"id field {id_field} of {self} is not an instance of Id")
```

Comment:
To avoid useless security check and simplify the condition (IMHO). The message can be improve too.

```suggestion
            if hasattr(self, 'id') and not isinstance(self.id, Id)):
                raise TypeError(f"{self.id!r} field of {self._name!} is not an instance of Fields.Id")
```